source/Timer.cpp

```diff
@@ -0,0 +1,346 @@
+/* Timer.cpp
+Copyright (c) 2023 by Timothy Collett
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "Timer.h"
+
+#include "DataNode.h"
+#include "DataWriter.h"
+#include "GameData.h"
+#include "Logger.h"
+#include "Planet.h"
+#include "PlayerInfo.h"
+#include "Random.h"
+#include "Ship.h"
+#include "StellarObject.h"
+#include "UI.h"
+
+using namespace std;
+
+namespace {
+	string TriggerToText(Timer::TimerTrigger trigger)
+	{
+		switch(trigger)
+		{
+			case Timer::TimerTrigger::TIMEUP:
+				return "on timeup";
+			case Timer::TimerTrigger::RESET:
+				return "on reset";
+			default:
+				return "unknown trigger";
+		}
+	}
```

Comment:
you can make this a const std::map

Reply:
This is an exact mirror of the way it's done in both NPC and Mission—should both of those be refactored to be `const std::map`s as well?

Reply:
Now I am unsure

Reply:
Let's leave this as is for now. We can refactor in the future if we want.